lib/improver/tests/utilities/test_cube_manipulation.py

```diff
@@ -527,6 +545,694 @@ def test_cubelist_different_var_names(self):
+        self.assertEqual(cubelist[0].coord("model").points[0],
+                         'Operational UKV Model Forecast')
+        self.assertArrayAlmostEqual(cubelist[1].coord("model_id").points,
+                                    np.array([100]))
+        self.assertEqual(cubelist[1].coord("model").points[0],
+                         'Operational Mogreps UK Model Forecast')
+        self.assertNotIn("title", cubelist[0].attributes.keys())
+        self.assertNotIn("title", cubelist[1].attributes.keys())
+        self.assertAlmostEquals(cubelist[0].attributes["grid_id"],
+                                cubelist[1].attributes["grid_id"])
+        self.assertEqual(cubelist[0].attributes["grid_id"],
+                         'ukx_standard_v1')
+        self.assertNotIn("history", cubelist[0].attributes.keys())
+        self.assertNotIn("history", cubelist[1].attributes.keys())
+
+    def test_strip_var_names(self):
+        """Test that the utility removes var names"""
+        cube1 = self.cube.copy()
+        cube2 = self.cube.copy()
+        cube1.coord("time").var_name = "time_0"
+        cube2.coord("time").var_name = "time_1"
+        cubelist = iris.cube.CubeList([cube1, cube2])
+        result = equalise_cubes(cubelist)
+        self.assertIsNone(cube1.coord("time").var_name)
+        self.assertIsNone(cube2.coord("time").var_name)
+
+    def test_coords_not_equalised_if_not_merging(self):
+        """Test that the coords are not equalised if not merging"""
+        cubelist = iris.cube.CubeList([self.cube_ukv, self.cube])
+        result = equalise_cubes(cubelist, merging=False)
+        self.assertEqual(len(result),
+                         len(cubelist))
+
+    def test_coords_are_equalised_if_merging(self):
+        """Test that the coords are equalised if merging"""
+        cubelist = iris.cube.CubeList([self.cube_ukv, self.cube])
+        result = equalise_cubes(cubelist)
+        self.assertEqual(len(result), 4)
+        self.assertAlmostEquals(result[3].coord('model_realization').points,
+                                102.0)
+
+
+class Test_equalise_cube_attributes(IrisTest):
+
+    """Test the equalise_cube_attributes utility."""
+
+    def setUp(self):
+        """Use temperature cube to test with."""
+        self.cube = set_up_temperature_cube()
+        self.cube_ukv = self.cube.extract(iris.Constraint(realization=1))
+        self.cube_ukv.remove_coord('realization')
+
+    def test_cubelist_history_removal(self):
+        """Test that the utility removes history attribute,
+        if they are different.
+        """
+        cube1 = self.cube.copy()
+        cube2 = self.cube.copy()
+        cube2.coord("time").points = 402195.0
+        cube1.attributes["history"] = "2017-01-18T08:59:53: StaGE Decoupler"
+        cube2.attributes["history"] = "2017-01-19T08:59:53: StaGE Decoupler"
+
+        cubelist = iris.cube.CubeList([cube1, cube2])
+
+        equalise_cube_attributes(cubelist)
```

Comment:
I'm not sure it matters hugely, but we should probably test what is returned by this function, rather than it's in-situ operation on the input.

Reply:
OK, I think it is worth noting that these functions update the original cubes and do not make a copy to work on. This is very useful from a memory and efficiency point of view but I suspect it may cause some unexpected results. We may need to revisit this in the future unless we feel strongly this is something we need to address now.